wger/core/api/serializers.py

```diff
@@ -16,14 +16,84 @@
 # along with Workout Manager.  If not, see <http://www.gnu.org/licenses/>.
 
 from rest_framework import serializers
-
+import re
+from django.contrib.auth.models import User
+from django.utils import translation
+from wger.config.models import GymConfig
+from rest_framework.authtoken.models import Token
+from django.shortcuts import get_object_or_404
 from wger.core.models import (
     UserProfile,
     Language,
     DaysOfWeek,
     License,
     RepetitionUnit,
-    WeightUnit)
+    WeightUnit,
+    UserModel
+)
+from wger.gym.models import GymUserConfig
+
+
+class UserApiSerializer(serializers.ModelSerializer):
+    """ Serializer to map to User model in relation to api user"""
+    api_key = serializers.CharField(required=True)
+    username = serializers.CharField(required=True)
+    email = serializers.EmailField(required=True)
+    password = serializers.CharField(required=True, min_length=8, write_only=True)
+
+    class Meta:
+        model = User
+        read_only_fields = ['api_key', 'username', 'password', 'email']
+        fields = ['api_key', 'username', 'password', 'email']
+
+    def validate(self, data):
+        check_email = User.objects.filter(email=data['email'])
+        user_check = User.objects.filter(
+            username=data['username'],
+            email=data['email'],
+            password=data['password']).exists()
+        if user_check:
+            raise serializers.ValidationError('user already exists')
+        if check_email.exists():
+            raise serializers.ValidationError("Email already in use.")
+        elif not re.match(r"^[a-zA-Z0-9_.-]+$", data['username']):
+            raise serializers.ValidationError('Invalid username provided')
+        elif not re.search(r"^(\w+\d+|\d+\w+)+$", data['password']):
+            raise serializers.ValidationError(
+                'Password should be a combination of Alphabets and Numbers')
+        return dict(data)
```

Comment:
Yes, it is but an OrderdDict which i force back to regular dict, to easily capture data at create method below. When i remove the dict() the create method fails, so that why i used it.